server/libs/userlibs.js

```diff
@@ -148,4 +149,100 @@ const logOut = function(request) {
     })
 }
 
-module.exports = {register, logIn, updateUser, logOut}
\ No newline at end of file
+const adminDeActivateUser = function(body) {
+    let id = body.uuid;
+    return new Promise((resolve,reject) => {
+        User.findOne({uuid : id}).then((user) => {
+            if (user) {
+                if (user.isAdmin) {
+                    let query = {
+                        deActivate : true,
+                        deletedAt : Date.now(),
+                        deletedBy : user.email
+                    }
+                    User.findOneAndUpdate({uuid : id, deActivate : false}, query, {new : true}).then((user) => {
+                        if(user) {
+                            resolve({
+                                status : statusCodes.successful,
+                                message : `User ${messages.deleted}`, 
+                                data : user
+                            });
+                        } else {
+                            reject({
+                                status : statusCodes.notFound,
+                                message : `User ${messages.notFound}`
+                            });
+                        }
+                    }).catch((error) => {
+                        reject({
+                            status : statusCodes.badRequest,
+                            data : error
+                        });
+                    })
+                } else {
+                    reject({
+                        status : statusCodes.forBidden,
+                        message : `User ${messages.notAdmin}`
+                    });
+                }
+            } else {
+                reject({
+                    status : statusCodes.notFound,
+                    message : `User ${messages.notFound}`
+                });
+            }
+        }).catch((error) => {
+            reject({
+                status : statusCodes.badRequest,
+                data : error
+            });
+        })
+    })
+}
+
+const userDelete = function(body) {
+    let id = body.uuid;
+    return new Promise((resolve,reject) => {
+        User.findOneAndDelete({uuid : id, deActivate : false}).then((user) => {
+            if (user) {
+                let decoded = {};
+                try {
+                    decoded = jwt.verify(body.token, secretKeys.tokenKey, process.env.JWT_SECRET);
```

Comment:
create one generic function for this decode and use it not every time you have to make it, Apply in all the code

Reply:
Done